pkg/pillar/cmd/zedagent/handleProfile.go

```diff
@@ -0,0 +1,261 @@
+		if localProfileServerIP != nil {
+			//defined ip
+			if config.Subnet.Contains(localProfileServerIP) {
+				res[localServerURL] = config.Gateway
+			}
+		} else {
+			//defined host, search in DNS records
+			for _, el := range config.DnsNameToIPList {
+				if el.HostName == localProfileServerHostname {
+					for _, ip := range el.IPs {
+						localServerURLReplaced := strings.Replace(localServerURL, localProfileServerHostname,
+							ip.String(), 1)
+						res[localServerURLReplaced] = config.Gateway
+						log.Functionf(
+							"prepareLocalProfileServerMap: will use url with replaced hostname: %s",
+							localServerURLReplaced)
+					}
+				}
+			}
+		}
+	}
+	return res, nil
+}
+
+// getLocalProfileConfig connects to local profile server to fetch current profile
+func getLocalProfileConfig(localServerURL string, iteration int, getconfigCtx *getconfigContext) error {
+
+	log.Tracef("getLocalProfileConfig(%s, %d)", localServerURL, iteration)
+
+	localServerMap, err := prepareLocalProfileServerMap(localServerURL, getconfigCtx)
+	if err != nil {
+		return fmt.Errorf("getLocalProfileConfig: prepareLocalProfileServerMap: %s", err)
+	}
+
+	if len(localServerMap) == 0 {
+		return fmt.Errorf(
+			"getLocalProfileConfig: cannot find any configured local subnets for localServerURL: %s",
+			localServerURL)
+	}
+
+	var errList []string
+	for link, srcIP := range localServerMap {
+		resp, contents, err := zedcloud.SendLocal(zedcloudCtx, link, srcIP, 0, nil)
+		if err != nil {
+			errList = append(errList, fmt.Sprintf("SendLocal: %s", err))
+			continue
+		}
+		if resp.StatusCode == http.StatusOK {
+			if err := validateProtoMessage(link, resp); err != nil {
+				// send something to ledmanager ???
+				errList = append(errList, fmt.Sprintf("validateProtoMessage: resp header error: %s", err))
+				continue
+			}
+			getconfigCtx.localProfileReceived = true
+			err := validateAndSetLocalProfile(contents, getconfigCtx)
+			if err != nil {
+				errList = append(errList, fmt.Sprintf("validateAndSetLocalProfile: %s", err))
+				continue
+			} else {
+				return nil
+			}
+		}
+		errList = append(errList, fmt.Sprintf("getLocalProfileConfig: wrong response status code: %d",
+			resp.StatusCode))
+	}
+	if !getconfigCtx.readSavedLocalProfile {
+		if err := readSavedLocalProfile(getconfigCtx); err != nil {
+			errList = append(errList, fmt.Sprintf("readSavedLocalProfile: %s", err))
+		} else {
+			return nil
+		}
+	}
+	return fmt.Errorf("getLocalProfileConfig: all attempts failed: %s", strings.Join(errList, ";"))
+}
+
+func writeReceivedProtoMessageLocalProfile(contents []byte) {
+	writeProtoMessage(savedLocalProfileFile, contents)
+}
+
+func parseProfile(ctx *getconfigContext, config *zconfig.EdgeDevConfig) {
+	log.Functionf("parseProfile start: globalProfile: %s currentProfile: %s",
+		ctx.globalProfile, ctx.currentProfile)
+	ctx.globalProfile = config.GlobalProfile
+	if ctx.localProfileServer != config.LocalProfileServer {
```

Comment:
Has this always been parsed/updated before parseProfile is called? Are there comments indicating that ordering constraint somewhere?

Reply:
This is the first place we set an option. Added the comment about do it before parsing apps on top of function.

Reply:
I'd put that order comment where this function is called to point out that it must be called before parseAppInst